content/courses/tokens-and-nfts/nfts-with-metaplex-core.md

```diff
@@ -0,0 +1,834 @@
+Keypairs, PublicKeys, and Connections. Fortunately, it's easy to convert between 
+web3.js and Umi equivalents.
+
+For more deetails, visit the [Metaplex Developer Docs](https://developers.metaplex.com/umi).
+
+#### Installing and setting up Umi
+
+First, create a new Umi instance. We can do this by either providing our own RPC 
+endpoint, or use the public facing Solana endpoints provided by the `clusterApiUrl` 
+method.
+
+```typescript
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import { clusterApiUrl } from "@solana/web3.js";
+
+const umi = createUmi(clusterApiUrl("devnet"));
+```
+
+Next, set the identity for your Umi instance (the keypair what will be used to sign 
+transactions) and load the necessary plugins, in this case, the `mplCore` plugin.
+
+```typescript
+import { mplCore } from "@metaplex-foundation/mpl-core";
+import { keypairIdentity } from "@metaplex-foundation/umi";
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import { getKeypairFromFile } from "@solana-developers/helpers";
+import { promises as fs } from "fs";
+import { clusterApiUrl } from "@solana/web3.js";
+
+const umi = createUmi(clusterApiUrl("devnet")).use(mplCore());
+
+// load keypair from local file system
+// See https://github.com/solana-developers/helpers?tab=readme-ov-file#get-a-keypair-from-a-keypair-file
+const localKeypair = await getKeypairFromFile();
+
+// convert to Umi compatible keypair
+const umiKeypair = umi.eddsa.createKeypairFromSecretKey(localKeypair.secretKey);
+
+// load the MPL metadata program plugin and assign a signer to our umi instance
+umi.use(keypairIdentity(umiKeypair));
+```
+
+#### Uploading images
+
+Before creating an Asset, you must prepare and upload any assets you plan to associate 
+with it. While this doesn't have to be an image, most Assets have an image associated 
+with them.
+
+Preparing and uploading an image involves converting to a buffer first. You can convert
+the file to a [generic file](https://developers.metaplex.com/umi/storage#generic-files) 
+using the `createGenericFile()` function and finally uploading it to the designated 
+Storage Driver.
+
+The `GenericFile` type allows Umi to support different file variations despite the 
+difference of browser files and local file system files i.e. those on your computer.
+
+In action, uploading an image named `random-image.png` from your computer would take the following steps:
+
+1. Reading the file using `readFile` into a buffer.
+
+2. Creating a generic file type with the files MIME Type from the buffer and filePath.
+
+3. Uploading file to designated storage provider.
+
+```typescript
+let filePath = "random-image.png";
+
+const buffer = await fs.readFile(filePath);
+let file = createGenericFile(buffer, filePath, {
+  // chose the correct file MIME type https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
+  contentType: "image/jpeg",
+});
+const [image] = await umi.uploader.upload([file]);
```

Comment:
You haven't configured an uploader here up to this point to be able to upload anything?